### ISSUE TITLE
Support for artifactory (java 17)

### ISSUE THREAD

**Issue by chepati** [open-telemetry/opentelemetry-java-instrumentation#9870]
### Is your feature request related to a problem? Please describe.

**TL;DR:**
Self-hosted (kubeadm) K8S cluster
Otel Operator 0.42.1 deployed via Helm chart
Artifactory OSS releases-docker.jfrog.io/jfrog/artifactory-oss:latest (running in Java 17 JVM, this is the default JVM)
Enabled java auto-instrumentation
Error:
_kubectl logs -n artifactory artifactory-server-deployment-7f9cbcff8b-8svnk artifactory-server
Preparing to run Artifactory in Docker
Running as uid=1030(artifactory) gid=1030(artifactory) groups=1030(artifactory)
Dockerfile for this image can found inside the container.
To view the Dockerfile: 'cat /docker/artifactory-oss/Dockerfile.artifactory'.
Copying Artifactory bootstrap files
 **is too old must be at least java 11**_

**Long version:**
Please find attached my manifest file.
[artifactory_otel.txt](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/13361796/artifactory_otel.txt)

And this is the auto-instrumented pod:
[artifactory_pod.txt](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/13362840/artifactory_pod.txt)

What can I do to make java autoinstrumentation work with Artifactory (java 17)?

### Describe the solution you'd like

I should be able to just auto-instrument Artifactory with the java agent and have no compatibility issues.

**Comment by laurit** [comment]
My guess is that the problem is in the way you are attaching the agent. You are probably adding the agent to the `JAVA_TOOL_OPTIONS` environment variable. This applies to all java processes, including the one that the artifactory startup script uses to detect java version. The extra output that the agent adds messes up the version detection in that script. There are multiple ways to solve this.
- Consult the documentation of whatever it is that you wish to run. Perhaps there is a different environment variable like `JAVA_OPTS` that is passed only to the process that you wish to run with the agent.
- Disable agent output. This can be done by adding -Dio.opentelemetry.javaagent.slf4j.simpleLogger.defaultLogLevel=off to jvm arguments. Instead of `off` you could also use `warn` as it is really the info level output that causes the issue.

**Comment by chepati** [comment]
@laurit : thanks for your reply.  I have attached a second file that describes the auto-instrumented (by the otel operator) artifactory pod.

Please let me know if I need to provide further info.

**Comment by laurit** [comment]
@chepati sorry, I don't know anything about kubernetes or otel operator. As described before I believe that this is a configuration issue rather than an agent issue.

**Comment by chepati** [comment]
We've narrowed it down to this:

$ java8 -jar opentelemetry-javaagent.jar
1.32.0

The agent prints its version and it is this version that confuses the artifactory jar into thinking it's being run under an obsolete version of java.

As a workaround we're starting artifactory by setting

JAVA_TOOL_OPTIONS="-Dio.opentelemetry.javaagent.slf4j.simplelogger.defaultLogLevel=warn"

However this will prevent us from later enabling debug mode if we need to view debug info.

Any way to suppress version printing for the java agent?

**Comment by laurit** [comment]
try `-Dio.opentelemetry.javaagent.slf4j.simpleLogger.log.io.opentelemetry.javaagent.tooling.VersionLogger=off`

**Comment by chepati** [comment]
Thanks, @laurit -- that did it!

**Comment by chepati** [comment]
Closing the issue.
